app.go

```diff
@@ -193,9 +198,43 @@ func New(opts ...Option) *App {
 		app.provide(p)
 	}
 	app.provide(func() Lifecycle { return app.lifecycle })
+
+	if app.err != nil {
+		app.logger.Printf("Error after options were applied: %v", app.err)
+		return app
+	}
+
+	if err := app.runInvokes(); err != nil {
+		app.err = err
+	}
+
 	return app
 }
 
+// runInvokes will execute invokes in order.
```

Comment:
That was the original wording, but I didn't want to use the word "provide" to describe part of the "invoke" functionality related to the order.